### PR TITLE
feat(rest): Add pagination to get clearing requests endpoint and fix 403 forbidden error.

### PR DESCRIPTION
**Pull request by akshitjoshii**
Added **pagination** for get all visible clearing requests endpoint and fixed 403 forbidden error for CRs created for projects with **"ME_AND_MODERATORS**" visibility.

Endpoint : http://localhost:8080/resource/api/clearingrequests?page=0&page_entries=2

closes : #2646 

image

### CODE REVIEW

**Comment by sameed20** [comment]
Test Successful

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java [hunk 1]
```diff
@@ -126,10 +119,14 @@ public ResponseEntity<EntityModel<ClearingRequest>> getClearingRequestByProjectI
...
-            Project projectWithClearingInfo = projectService.getClearingInfo(project, sw360User);
-            restControllerHelper.addEmbeddedReleaseDetails(halClearingRequest, projectWithClearingInfo);
-            restControllerHelper.addEmbeddedProject(halClearingRequest, project, true);
+            try{
+                Project project = projectService.getProjectForUserById(clearingRequest.getProjectId(), sw360User);
+                Project projectWithClearingInfo = projectService.getClearingInfo(project, sw360User);
+                restControllerHelper.addEmbeddedReleaseDetails(halClearingRequest, projectWithClearingInfo);
+                restControllerHelper.addEmbeddedProject(halClearingRequest, project, true);
+            }catch (Exception e){
+                return null;
```
Comment (sameed20) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java:128]: Please add a log statement.

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java [hunk 2]
```diff
@@ -9,15 +9,10 @@
...
-import static org.eclipse.sw360.datahandler.common.WrappedException.wrapTException;
 import static org.springframework.hateoas.server.mvc.WebMvcLinkBuilder.linkTo;
 
 import java.net.URISyntaxException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.List;
-import java.util.Set;
-import java.util.TreeSet;
+import java.util.*;
```
Comment (sameed20) [inline rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java:15]: Please remove this import **import java.util.*;**

File: rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java (hunk 1 above)
Comment (akshitjoshii) [reply rest/resource-server/src/main/java/org/eclipse/sw360/rest/resourceserver/clearingrequest/ClearingRequestController.java:136]: added log statement

**Comment by akshitjoshii** [comment]
@sameed20, addressed the review comments.

**Comment by sameed20** [comment]
LGTM

**Review by GMishx — APPROVED** [review]
Reviewed by @sameed20 here: https://github.com/eclipse-sw360/sw360/pull/2658#issuecomment-2453908829

**Pull request merged**